src/uniswap.act.md

```diff
@@ -734,6 +734,100 @@ calls
     UniswapV2Pair.balanceOf
 ```
 
+### Swap
+
+```act
+behaviour swap of UniswapV2Pair
+interface swap(uint amount0Out, uint amount1Out, address to, bytes calldata data)
+
+for all
+
+    Reserve0           : uint112
+    Reserve1           : uint112
+    Balance0           : uint112
+    Balance1           : uint112
+    BlockTimestampLast : uint32
+    Token0             : address UniswapV2Pair
+    Token1             : address UniswapV2Pair
+    Price0             : uint256
+    Price1             : uint256
+    DstBal0            : uint256
+    DstBal1            : uint256
+    BlockTimestampLast : uint32
+
+storage
+
+    reserve0_reserve1_blockTimestampLast |-> #WordPackUInt112UInt112UInt32(Reserve0, Reserve1, BlockTimestampLast) => #WordPackUInt112UInt112UInt32(Balance0 - amount0Out, Balance1 - amount1Out, TIME)
+    token0   |-> Token0
+    token1   |-> Token1
+    lockState |-> 1
+    price0CumulativeLast |-> Price0 => #if TIME - BlockTimestampLast =/= 0 #then ((((pow112 * Reserve1) / Reserve0) * (TIME - BlockTimestampLast)) + Price0) #else Price0 #fi
+    price1CumulativeLast |-> Price1 => #if TIME - BlockTimestampLast =/= 0 #then ((((pow112 * Reserve0) / Reserve1) * (TIME - BlockTimestampLast)) + Price1) #else Price1 #fi
```

Comment:
I think you're missing the checks for non zero reserves before updating the price accumulators here

Reply:
actually i think its not necessary, its implied from the other assumptions and the behaviour

Reply:
but maybe i will need to add this with allowing some stuff to overflow where the nonzero constraints came from, i will make a run and report back

Reply:
You can check the `sync` and `mint` specs for examples of specs that deal with overflow in `_update`.